tests/Microsoft.Identity.Test.Unit.net45/PublicApiTests/AcquireTokenSilentTests.cs

```diff
@@ -147,49 +148,57 @@ public void AcquireTokenSilentScopeAndUserOverloadDefaultAuthorityTest()
         {
+            string instanceMetadataJson = File.ReadAllText(ResourceHelper.GetTestResourceRelativePath("CustomInstanceMetadata.json"));
+
             using (var httpManager = new MockHttpManager())
             {
-                PublicClientApplication app = PublicClientApplicationBuilder.Create(MsalTestConstants.ClientId)
-                                                                            .WithAuthority(new Uri(MsalTestConstants.AuthorityGuestTenant), true)
-                                                                            .WithHttpManager(httpManager)
-                                                                            .WithTelemetry(new TraceTelemetryConfig())
-                                                                            .BuildConcrete();
-
-                var tokenCacheHelper = new TokenCacheHelper();
-                tokenCacheHelper.PopulateCache(app.UserTokenCacheInternal.Accessor);
-                app.UserTokenCacheInternal.Accessor.DeleteAccessToken(
-                    new MsalAccessTokenCacheKey(
-                        MsalTestConstants.ProductionPrefNetworkEnvironment,
-                        MsalTestConstants.Utid,
-                        MsalTestConstants.UserIdentifier,
-                        MsalTestConstants.ClientId,
-                        MsalTestConstants.ScopeForAnotherResourceStr));
-
-                httpManager.AddInstanceDiscoveryMockHandler();
-                httpManager.AddMockHandlerForTenantEndpointDiscovery(MsalTestConstants.AuthorityGuestTenant);
-
-                httpManager.AddMockHandler(
-                    new MockHttpMessageHandler()
-                    {
-                        ExpectedMethod = HttpMethod.Post,
-                        ResponseMessage = MockHelpers.CreateSuccessTokenResponseMessage(
-                            MsalTestConstants.UniqueId,
-                            MsalTestConstants.DisplayableId,
-                            MsalTestConstants.Scope.ToArray())
-                    });
-
-                Task<AuthenticationResult> task = app
-                    .AcquireTokenSilent(
-                        MsalTestConstants.Scope.ToArray(),
-                        new Account(MsalTestConstants.UserIdentifier, MsalTestConstants.DisplayableId, null))
-                    .ExecuteAsync(CancellationToken.None);
-
-                AuthenticationResult result = task.Result;
-                Assert.IsNotNull(result);
-                Assert.AreEqual(MsalTestConstants.DisplayableId, result.Account.Username);
-                Assert.AreEqual(MsalTestConstants.Scope.AsSingleString(), result.Scopes.AsSingleString());
+               
+
+                var app = PublicClientApplicationBuilder.Create(MsalTestConstants.ClientId)
+                                                                           .WithAuthority(new Uri(MsalTestConstants.AuthorityGuestTenant), false)
```

Comment:
Why false?

Reply:
It doesn't make sense to set validate metadata = true and to have this set. So they'll have to set validate flag to false, otherwise I throw, i.e. these 2 settings are mutually exclusive.  I think this will require developers to understand if they really need to do this, read the docs etc.

Reply:
CC @jmprieur to whom I have discussed about this, but I believe it is an important detail.

Reply:
@henrik-me - the exception message is: 

_"You have configured custom instance metadata, but the validateAuthority flag is set to true. These are mutually exclusive. Set the validateAuthority flag to false. See https://aka.ms/msal-net-custom-instance-metadata for more details."_